adafruit_mpr121.py

```diff
@@ -79,71 +97,96 @@
 
 class MPR121_Channel:
     # pylint: disable=protected-access
-    """Helper class to represent a touch channel on the MPR121. Not meant to
-    be used directly."""
+    """Represents a single channel on the touch sensor.
+
+    Not meant to be used directly.
+    """
+
+    _mpr121: MPR121
+    _channel: int
```

Comment:
I'm not familiar with this syntax that is typing things outside of function arguments and returns. 

We'll need to make sure that mpy-cross and the microcontrollers can handle this correctly. I don't know if we've had any other libraries include types like this yet.

Reply:
I add these for readability, static type checkers, and consistency because many other languages have some form of this syntax. Let me know what those tests yield and I'll make the change if requested.

Reply:
I tested the mpy-cross build and basic functionality of the library with a Feather ESP32S2 TFT and https://www.adafruit.com/product/4830. 

As far as I can tell both mpy-cross and the microcontroller are not having any issue with the types. We should be able to keep these.

Reply:
How does the memory usage look? Any increase is size between these being there and not? I'm curious if the compiler tosses these out like comments or docstrings to save space.

Reply:
I am unsure how they get treated by mpy-cross or the interpreter. I tested with this:
```
import gc
import time
import board
import busio

# Import MPR121 module.
import adafruit_mpr121

# Create I2C bus.
i2c = busio.I2C(board.SCL, board.SDA)

# Create MPR121 object.
mpr121 = adafruit_mpr121.MPR121(i2c)
print(gc.mem_alloc())
```
On a Feather S2 TFT and it showed `15296` used.

I tried testing the currently released one similarly, but my device has a one of dev build and the mpy files are not compatible right now.

I don't see anything alarming with usage, and the rest of these changes are looking good to me now. I don't think we need to hold up merging to test the specific usage.